services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/volumes_resolver.py

```diff
@@ -82,30 +86,79 @@ def _volume_name(cls, path: Path) -> str:
         return f"{path}".replace(os.sep, "_")
 
     @classmethod
-    def source(cls, compose_namespace: str, path: Path) -> str:
-        return f"{compose_namespace}{cls._volume_name(path)}"
+    def source(cls, path: Path, node_uuid: NodeID, run_id: RunID) -> str:
+        """
+        Guarantees that the volume name is unique between runs while also
+        taking into consideration the limit for the volume name's length
+        (255 characters).
+
+        Example:
+        >>> source(
+            path="/home/user/a-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-long-home-path/workspace",
+            node_uuid="17f5b46d-505c-489e-9818-f6d294c892d9",
+            run_id="a5d2a8bd-eaef-4d7b-8dd9-1ae4e408f3c3"
+        )
+        dyv_f728b4fa-4248-4e3a-8a5d-2f346baa9455_ecapskrow_htap-emoh-gnol-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-a_resu_emoh_e3e70682-c209-4cac-a29f-6fb
```

Comment:
sorry but I do not understand your example here...
where is the node_uuid?

Reply:
The node_uuid gets appended at the end and is cut off.  Let me properly update it, I forgot to change the entire comment.

Reply:
Will look something like this
```python
>>> source(
    path="/home/user/a-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-long-home-path/workspace",
    node_uuid="17f5b46d-505c-489e-9818-f6d294c892d9",
    run_id="f728b4fa-4248-4e3a-8a5d-2f346baa9455"
)
dyv_f728b4fa-4248-4e3a-8a5d-2f346baa9455_ecapskrow_htap-emoh-gnol-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-a_resu_emoh_17f5b46d-505c-489e-9818-f6d
```

Reply:
I thought we said it's useful to be able to search via node_uuid. I'm confused.. but ok..

Reply:
Now that I read this. I managed to change it and make it readable.
```python
>>> source(
    path="/home/user/a-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-very-long-home-path/workspace",
    node_uuid="e3e70682-c209-4cac-a29f-6fbed82c07cd",
    run_id="f728b4fa-4248-4e3a-8a5d-2f346baa9455"
)
dyv_f728b4fa-4248-4e3a-8a5d-2f346baa9455_e3e70682-c209-4cac-a29f-6fbed82c07cd_ecapskrow_htap-emoh-gnol-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yrev-yr
```